Tests/AppTests/Models/PackageTests.swift

```diff
@@ -7,45 +7,55 @@ import XCTest
 final class PackageTests: XCTestCase
 {
   var app: Application!
-  var conn: PostgreSQLConnection!
+  var database: PostgreSQLConnection!
 
   override func setUp()
   {
     try! Application.reset()
     app = try! Application.testable()
-    conn = try! app.newConnection(to: .psql).wait()
+    database = try! app.newConnection(to: .psql).wait()
+
+    // Create a couple of test packages that can be relied on during a test
+    createTestPackage(from: "https://github.com/vapor/vapor.git")
+    createTestPackage(from: "https://github.com/Alamofire/Alamofire.git")
   }
 
   override func tearDown()
   {
-    conn.close()
+    database.close()
     try? app.syncShutdownGracefully()
   }
 
   func testPackageCreationWithUrl() throws
   {
-    let url = URL(string: "https://example.com/")
+    let url = URL(string: "https://example.com/")!
     let package = Package(url: url)
     XCTAssertEqual(package.url, url)
   }
 
-  func testPackageCreationWithUrlString() throws
+  func testFindPackageByURL() throws
   {
-    let urlString = "https://example.com/"
-    let package = Package(urlString: urlString)
-    XCTAssertEqual(package.url, URL(string: urlString))
+    let existingUrl = URL(string: "https://github.com/Alamofire/Alamofire.git")!
+    let existingPackage = try Package.findByUrl(on: database, url: existingUrl).wait()
+    XCTAssertEqual(existingPackage.url, existingUrl)
+
+    let nonExistentUrl = URL(string: "https://github.com/apple/shiny.git")!
+    let nonExistentPackage = try Package.findByUrl(on: database, url: nonExistentUrl).wait()
+    XCTAssertNil(nonExistentPackage)
```

Comment:
This test fails with:

```
<unknown>:0: error: -[AppTests.PackageTests testFindPackageByURL] : failed: caught error: The operation couldn’t be completed. (App.PackageError error 0.)
```

Presumably, that's the equivalent of a record not found type error. Maybe that's good enough that this second assertion isn't needed?

Reply:
You do `XCTAssetThrows` to ensure it throws the correct error?

Reply:
> Presumably, that's the equivalent of a record not found type error.

5 minutes ago Dave is an idiot. It was actually throwing my very own PackageError.recordNotFound error which I defined and told it to throw 😬

Fixed up this test to correctly test for that.